Analisis ini bertujuan untuk memahami beberapa aspek penting dari aktivitas penjualan.


1.   Produk yang paling diminati pelanggan
2.   Kontribusi keuntungan dari setiap produk
3. Performa penjualan berdasarkan lokasi toko
4. Karakteristik pelanggan
5. Pola waktu transaksi
6. Tingkat kepuasan pelanggan


Dari analisis data ini diharapkan diperoleh beberapa insight, yaitu:


*   Produk dengan penjualan tertinggi dan kontribusi keuntungan terbesar
*   Cabang atau kota dengan performa penjualan terbaik

*   Pola perilaku pelanggan berdasarkan gender dan tipe pelanggan
*   Metode pembayaran yang paling banyak digunakan pelanggan

*   Waktu transaksi paling ramai dalam periode tertentu
*   Kategori produk atau cabang dengan tingkat kepuasan pelanggan tertinggi

#Persiapan dan Cleaning Data

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

Load Dataset

In [ ]:
df = pd.read_csv("supermarket_sales.csv")
df.head()

,Invoice ID,Branch,City,Customer type,Gender,Product line,Unit price,Quantity,Tax 5%,Total,Date,Time,Payment,cogs,gross margin percentage,gross income,Rating
0,750-67-8428,A,Yangon,Member,Female,Health and beauty,74.69,7,26.1415,548.9715,1/5/2019,13:08,Ewallet,522.83,4.761905,26.1415,9.1
1,226-31-3081,C,Naypyitaw,Normal,Female,Electronic accessories,15.28,5,3.8200,80.2200,3/8/2019,10:29,Cash,76.40,4.761905,3.8200,9.6
2,631-41-3108,A,Yangon,Normal,Male,Home and lifestyle,46.33,7,16.2155,340.5255,3/3/2019,13:23,Credit card,324.31,4.761905,16.2155,7.4
3,123-19-1176,A,Yangon,Member,Male,Health and beauty,58.22,8,23.2880,489.0480,1/27/2019,20:33,Ewallet,465.76,4.761905,23.2880,8.4
4,373-73-7910,A,Yangon,Normal,Male,Sports and travel,86.31,7,30.2085,634.3785,2/8/2019,10:37,Ewallet,604.17,4.761905,30.2085,5.3


Melihat Struktur Dataset

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

In [ ]:
df.isna().sum()

,0
Invoice ID,0
Branch,0
City,0
Customer type,0
Gender,0
Product line,0
Unit price,0
Quantity,0
Tax 5%,0
Total,0


Struktur dataset menunjukkan bahwa data memiliki 1000 baris dan 17 kolom (variabel). Dari keseluruhan variabel tersebut, tidak terdapat nilai kosong. Berdasarkan tipe datanya, dataset terdiri dari 7 variabel bertipe float64, 1 variabel bertipe int64, dan 9 variabel bertipe object. Variabel bertipe numerik seperti Unit price, Quantity, Tax 5%, Total, cogs, gross margin percentage, gross income, dan Rating merepresentasikan nilai kuantitatif yang berkaitan dengan transaksi penjualan dan keuntungan. Sementara itu, variabel bertipe object seperti Invoice ID, Branch, City, Customer type, Gender, Product line, Date, Time, dan Payment menggambarkan informasi kategorikal yang menjelaskan identitas transaksi, karakteristik pelanggan, jenis produk, serta metode pembayaran.

Mengubah Format Tanggal

In [ ]:
df['Date'] = pd.to_datetime(df['Date'])

Memisahkan Komponen Waktu

In [ ]:
df['Year'] = df['Date'].dt.year
df['Month'] = df['Date'].dt.month
df['Day'] = df['Date'].dt.day

In [ ]:
df['Hour'] = pd.to_datetime(df['Time']).dt.hour

/tmp/ipykernel_406/3352662874.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['Hour'] = pd.to_datetime(df['Time']).dt.hour


In [ ]:
df['Sales_per_item'] = df['Unit price'] * df['Quantity']

Menyimpan data yang sudah dibersihkan

In [ ]:
df.to_csv("Data_Cleaned.csv", index=False)

#Explorasi Data Analys

##Produk Paling Laku

In [ ]:
df.groupby('Product line')['Quantity'].sum().sort_values(ascending=False)

,Quantity
Product line,
Electronic accessories,971
Food and beverages,952
Sports and travel,920
Home and lifestyle,911
Fashion accessories,902
Health and beauty,854


Berdasarkan jumlah produk yang terjual pada setiap kategori, Electronic accessories merupakan produk dengan penjualan tertinggi dengan total 971 unit. Hal ini menunjukkan bahwa kategori tersebut merupakan produk paling diminati pelanggan dibandingkan kategori lainnya. Selanjutnya, kategori Food and beverages berada pada posisi kedua dengan 952 unit, diikuti oleh Sports and travel dengan 920 unit.
Secara keseluruhan, hasil ini menunjukkan bahwa produk yang berkaitan dengan **Electronic accessories** dan **kebutuhan konsumsi** memiliki tingkat permintaan yang lebih tinggi dibandingkan kategori produk lainnya.

##Produk dengan Keuntungan Tertinggi

In [ ]:
df.groupby('Product line')['gross income'].sum().sort_values(ascending=False)

,gross income
Product line,
Food and beverages,2673.5640
Sports and travel,2624.8965
Electronic accessories,2587.5015
Fashion accessories,2585.9950
Home and lifestyle,2564.8530
Health and beauty,2342.5590


Berdasarkan total gross income pada setiap kategori produk, Food and beverages merupakan kategori yang memberikan keuntungan terbesar dengan nilai sekitar 2673.56. Hal ini menunjukkan bahwa produk pada kategori tersebut memberikan kontribusi keuntungan paling tinggi bagi perusahaan dibandingkan kategori lainnya. Selanjutnya, kategori Sports and travel menempati posisi kedua dengan keuntungan sekitar 2624.90, diikuti oleh Electronic accessories sebesar 2587.50.
Secara keseluruhan, hasil ini menunjukkan bahwa meskipun beberapa kategori memiliki jumlah penjualan tinggi, kategori **Food and beverages** memberikan kontribusi keuntungan yang paling besar terhadap pendapatan perusahaan.


##Performa penjualan berdasarkan lokasi toko

In [ ]:
df.groupby('Branch')['Total'].sum()

,Total
Branch,
A,106200.3705
B,106197.6720
C,110568.7065


In [ ]:
df.groupby('City')['Total'].sum()

,Total
City,
Mandalay,106197.6720
Naypyitaw,110568.7065
Yangon,106200.3705


Berdasarkan total penjualan pada setiap cabang, Branch C memiliki performa penjualan terbaik dengan total penjualan sekitar 110.568,71. Sementara itu, Branch A dan Branch B memiliki nilai penjualan yang hampir sama, masing-masing sekitar 106.200,37 dan 106.197,67. Jika dilihat berdasarkan lokasi kota, hasilnya juga menunjukkan pola yang serupa karena setiap cabang berada pada kota yang berbeda. Naypyitaw menjadi kota dengan total penjualan tertinggi yaitu sekitar 110.568,71, diikuti oleh Yangon dengan 106.200,37, dan Mandalay dengan 106.197,67.

Secara keseluruhan, dapat disimpulkan bahwa cabang yang berada di **Naypyitaw** memiliki performa penjualan paling baik, sedangkan cabang di Yangon dan Mandalay memiliki tingkat penjualan yang relatif seimbang.

##Segmentasi Pelanggan

In [ ]:
df.groupby('Customer type')['Total'].mean()

,Total
Customer type,
Member,327.791305
Normal,318.122856


In [ ]:
df['Payment'].value_counts()

,count
Payment,
Ewallet,345
Cash,344
Credit card,311


Berdasarkan segmentasi pelanggan, rata-rata nilai transaksi pelanggan member sedikit lebih tinggi dibandingkan pelanggan normal. Pelanggan member memiliki rata-rata total transaksi sekitar 327.79, sedangkan pelanggan normal memiliki rata-rata transaksi sekitar 318.12. Hal ini menunjukkan bahwa pelanggan member cenderung melakukan pembelian dengan nilai transaksi yang lebih besar. Selain itu, jika dilihat dari metode pembayaran, pelanggan paling banyak menggunakan E-wallet dengan 345 transaksi, diikuti oleh cash sebanyak 344 transaksi, dan credit card sebanyak 311 transaksi. Hal ini menunjukkan bahwa metode pembayaran digital dan tunai merupakan pilihan utama pelanggan dalam melakukan transaksi.

Secara keseluruhan, pelanggan member memberikan kontribusi transaksi yang sedikit lebih tinggi, sementara metode pembayaran E-wallet menjadi metode yang paling sering digunakan oleh pelanggan.

##Pola Waktu Tranksaksi

In [ ]:
df.groupby('Hour')['Total'].sum()

,Total
Hour,
10,31421.4810
11,30377.3295
12,26065.8825
13,34723.2270
14,30828.3990
15,31179.5085
16,25226.3235
17,24445.2180
18,26030.3400


Berdasarkan pola waktu transaksi, penjualan terjadi pada rentang waktu pukul 10.00 hingga 20.00. Total transaksi tertinggi terjadi pada pukul 19.00 dengan nilai sekitar 39,699.51, yang menunjukkan bahwa aktivitas pembelian pelanggan paling ramai terjadi pada malam hari.

Selain itu, transaksi juga cukup tinggi pada pukul 13.00 dengan total sekitar 34,723.23, serta pada pukul 10.00 dan 15.00 yang masing-masing mencapai lebih dari 31,000. Sementara itu, aktivitas transaksi relatif lebih rendah pada pukul 20.00 dengan total sekitar 22,969.53.

Secara keseluruhan, pola ini menunjukkan bahwa waktu sore hingga malam hari merupakan periode dengan aktivitas transaksi paling tinggi, sehingga waktu tersebut menjadi momen penting bagi toko untuk mengoptimalkan pelayanan dan strategi penjualan.


##Tingkat Kepuasan Pelanggan

In [ ]:
df.groupby('Branch')['Rating'].mean()

,Rating
Branch,
A,7.027059
B,6.818072
C,7.072866


Berdasarkan rata-rata rating pelanggan pada setiap cabang, secara umum tingkat kepuasan pelanggan berada pada nilai sekitar **7 dari 10**, yang menunjukkan bahwa pengalaman berbelanja pelanggan tergolong cukup baik.

Di antara ketiga cabang, **Branch B** memiliki rata-rata rating paling rendah yaitu sekitar **6.82**.

Hasil ini menunjukkan bahwa **Branch B masih memiliki peluang untuk meningkatkan kualitas layanan agar kepuasan pelanggan dapat lebih optimal**.


#Strategi Bisnis

Dari berbagai insight yang diperoleh, dapat dirumuskan beberapa strategi yang dapat diterapkan untuk mendorong pertumbuhan serta mengoptimalkan kinerja bisnis:


1.  **Optimalisasi Portofolio Produk**: Manfaatkan popularitas "Electronic accessories" sebagai penarik pelanggan, dan secara strategis dorong penjualan "Food and beverages" yang memiliki margin keuntungan lebih tinggi melalui bundling atau penempatan produk yang cerdas.

2.  **Replikasi Keberhasilan Cabang**: Pelajari secara mendalam faktor-faktor kunci keberhasilan "Cabang C di Naypyitaw". Identifikasi praktik terbaik dalam operasional, pemasaran, dan layanan pelanggan yang dapat direplikasi ke "Cabang A di Yangon" dan "Cabang B di Mandalay" untuk meningkatkan performa penjualan mereka.

3.  **Penguatan Loyalitas Pelanggan**: Tingkatkan investasi dalam program keanggotaan untuk "pelanggan Member" guna mendorong nilai transaksi yang lebih tinggi.

4.  **Manajemen Operasional Berbasis Waktu**: Optimalkan penjadwalan staf, alokasi sumber daya, dan aktivitas promosi sesuai dengan "pola waktu transaksi puncak", terutama pada pukul 19.00 dan 13.00, untuk memaksimalkan pengalaman pelanggan dan potensi penjualan. Pertimbangkan penawaran khusus pada jam-jam tersebut.

5.  **Peningkatan Kualitas Layanan Berbasis Data**: Jadikan "Cabang B" sebagai fokus utama untuk program peningkatan kualitas layanan. Lakukan audit layanan, pelatihan staf, atau "mystery shopping" untuk mengidentifikasi dan mengatasi akar penyebab "rating kepuasan yang lebih rendah".

In [ ]:
df.to_csv("Data_Cleaned.csv", index=False)